include/exadg/operators/solution_transfer.h

```diff
@@ -0,0 +1,147 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_OPERATORS_SOLUTION_TRANSFER_H
+#define INCLUDE_EXADG_OPERATORS_SOLUTION_TRANSFER_H
+
+// deal.II
+#include <deal.II/distributed/solution_transfer.h>
+#include <deal.II/distributed/tria.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/numerics/solution_transfer.h>
+
+namespace ExaDG
+{
+template<int dim, typename VectorType>
+class SolutionTransfer
+{
+public:
+  /*
+   * Constructor.
+   */
+  SolutionTransfer(dealii::DoFHandler<dim> const & dof_handler_in)
+  {
+    dof_handler = &dof_handler_in;
```

Comment:
note that we will change this to std::vectors of `DoFHandler`s and `std::vector<VectorType *>` in the future to enforce a correct sequence of interpolation. This will be needed for the combination with ALE already (grid velocities and coords etc))